src/Planner/PlannerJoinTree.cpp

```diff
@@ -644,6 +645,37 @@ JoinTreeQueryPlan buildQueryPlanForTableExpression(QueryTreeNodePtr table_expres
         {
             if (!select_query_options.only_analyze)
             {
+                auto storage_merge_tree = std::dynamic_pointer_cast<MergeTreeData>(storage);
+                if (storage_merge_tree && query_context->canUseParallelReplicasOnInitiator()
+                    && settings.parallel_replicas_min_number_of_rows_per_replica > 0)
+                {
+                    /// This is trash
+                    /// It uses the old InterpreterSelectQuery to do the estimation of how many rows will be read
```

Comment:
Oh oh, probably better not to support automatic decision at all for new Analyzer for now?

Reply:
I spent a few days trying to implement this "properly" without success (either you don't have access to the parts, or don't have access to the final prewhere optimization, or the connection and remote queries are already setup), and in the end this was the only way I conceived possible without having to rethink how the analyzer works completely. There might be a better way I didn't see.

In the end I preferred the path of having a bad implementation for the analyzer that it worked (vs the feature not working) so people can start playing with parallel replicas and see what improvements are needed the most. But I don't have a strong opinion; it could be replaced by a warning or an exception (if using analyzer + settings.parallel_replicas_min_number_of_rows_per_replica +  parallel_replicas=2) instead.

Reply:
Let's ask @novikd or @KochetovNicolai about their opinion :)

Reply:
@KochetovNicolai @novikd Can you please have a look? I don't mind going either way but it'd be great to have this change to do some testing on the original interpreter and later improve the analyzer to be able to support this kind of decisions better.

Reply:
Kind reminder this is pending a decision to be unblocked.

Reply:
It's getting cold out here. It'd be great to have a decision so this can be used in different environments and see how important it is to have it in the analyzer.

Reply:
Yes. The PR uses the primary key analysis to decide how many parallel replicas (or disable it) before the whole setup happens, and that's not available here. So ideally yes, #52762 should make it possible to do this process in the analyzer in a simpler way.

Reply:
The PR was merged and reverted. The new PR introducing filters for analysis in the analyzer is in https://github.com/ClickHouse/ClickHouse/pull/53792

Reply:
Once https://github.com/ClickHouse/ClickHouse/pull/53792 has been merged the PR could be implemented on top of the analyzer